core/src/avm1/test_utils.rs

```diff
@@ -44,7 +44,18 @@ macro_rules! test_method {
                         $(
                             let args: Vec<Value> = vec![$($arg.into()),*];
                             let ret = crate::avm1::object::TObject::call_method(&object, name, &args, activation, crate::avm1::function::ExecutionReason::Special)?;
-                            assert_eq!(ret, $out.into(), "{:?} => {:?} in swf {}", args, $out, version);
+                            let out = $out.into();
+
+                            $(
+                                fn round_to_digits(x: f64) -> f64 {
+                                    let factor = f64::powi(10.0, $digits);
+                                    (x * factor).round() / factor
+                                }
+                                let ret = round_to_digits(ret.coerce_to_f64(activation)?);
+                                let out = round_to_digits(out);
+                            )?
+
+                            assert_eq!(ret, out, "{:?} => {:?} in swf {}", args, $out, version);
```

Comment:
IMO `@digits` is a bit over-engineered here and we should just use `@delta` and assert `abs(expected - actual) < delta`, as that's the standard way of comparing FP numbers.

`@digits` compares x digits after the decimal point, but the precision of floating point numbers is not defined for the fractional part, but for the whole number. A better approach would be to compare x significant digits. Also `@digits` uses base 10 instead of base 2, which is kinda arbitrary and unnecessarily affects mantissa, but I'm not sure if it can cause any problems here.

Reply:
> IMO `@digits` is a bit over-engineered here 

Still not as over-engineered as `@transform(|x| (x * 1e12).round() / 1e12)` could have been! :V

> we should just use `@delta`

How about `@epsilon`? IIRC "delta" is more of a domain thing than a value thing (in functional analysis at least).

>  compares x digits after the decimal point, but the precision of floating point numbers is not defined for the fractional part, but for the whole number.

Yeah, fair, I know - this was just a quickly thrown together late night hack.

Reply:
> uses base 10 instead of base 2, which is kinda arbitrary and unnecessarily affects mantissa

Can you suggest a simple and clear way to write a binary exponent `f64` literal?

Reply:
> How about `@epsilon`? IIRC "delta" is more of a domain thing than a value thing (in functional analysis at least).

Sure, seems alright!

> Can you suggest a simple and clear way to write a binary exponent `f64` literal?

Usually languages support hexadecimal floating point literals for this purpose, but I don't think Rust supports it (seems like it's a deliberate decision, but I don't know the reasoning). Currently you would have to convert the desired fp number to decimal and use that. There are alternatively some crates that can parse hex fp strings, or you could write the value as an expression.

Nonetheless, I was referring to mapping the original value (and potentially losing accuracy due to 5 not being a power of 2). For epsilon that should not matter, so we're safe here.